pkg/cincinnati/cincinnati.go

```diff
@@ -20,12 +20,13 @@ const (
 // Client is a Cincinnati client which can be used to fetch update graphs from
 // an upstream Cincinnati stack.
 type Client struct {
-	id uuid.UUID
+	id        uuid.UUID
+	transport *http.Transport
 }
 
 // NewClient creates a new Cincinnati client with the given client identifier.
-func NewClient(id uuid.UUID) Client {
-	return Client{id: id}
+func NewClient(id uuid.UUID, transport *http.Transport) Client {
```

Comment:
This is pretty broad additon.. 
I think a restricted proxy information and creating a transport with that is more ideal.